cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -198,8 +200,7 @@ public String commitAndPush(CommitMeta commitMeta, Path fileChanged)
     }
 
     try {
-      TreeWalk walk = TreeWalk.forPath(git.getRepository(), fileChanged.toString(), commit.getTree());
-      return walk.getObjectId(0).getName();
+      return getFileHash(fileChanged, commit);
     } catch (IOException e) {
       LOG.warn(String.format("Failed to get the fileHash for file: %s", fileChanged), e);
```

Comment:
nit: unrelated to this change, but it's weird we need to introduce such a use case and then handle it in all the callers. Is it something we really expect? If not, I'd just propagate the exception as non-checked